### ISSUE TITLE
Make it more obvious which prereq of a building are met

### ISSUE THREAD

**Issue by DarthGandalf** [ihhub/fheroes2#7276]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

I want to build a building in the city. It has several requirements: e.g. for the Mausoleum I need to build Mage Guild level 1 first. And I already built some of these required buildings. It would be nice to show visually in the list of requirements which of them are met already (with a checkmark or something). Same for resources: if I don't have enough crystals, but have enough wood, it should mark them as such, without requiring me looking at numbers back and forth until I see which of them is smaller than the other number.

### Describe the possible solution

Cross out the text with the required building, or add a checkbox next to it.

Resources - maybe show in red color the number of resource which I don't have enough of.

UI design of this is debatable :) And I'm not a good designer

**Comment by ihhub** [comment]
Closing this issue as duplicated of 5300.

**Comment by lisnenko** [comment]
@ihhub, I don't get it. Is it completed or not? 🤔
You closed it as completed on Aug 21, 2023 and reopened it today. 🤯

**Comment by ihhub** [comment]
@lisnenko , your issue is a duplicate of this one. No need to have more than one issue.

**Comment by lisnenko** [comment]
I assume this improvement is still not finished. More than 2 years. Okay. I am not familiar with capacities and priorities of this project.

**Comment by ihhub** [comment]
No, it's not as this is a very low "nice to have" improvement. The team has a lot of other tasks to work on as well.

**Comment by lisnenko** [comment]
I am sure it wouldn’t hurt to post here my suggestions on UI from https://github.com/ihhub/fheroes2/issues/10344

Highlight the resources that player does not have in red\bold red.
I mean to change the color of the font, that is used to show a quantity of a particular resource.
Other options: Glowing frame, glowing outline, strike them horizontally, diagonally, bold them, circle them, underline them.
Do something similar also with structures' dependencies?

**Comment by zenseii** [comment]
@lisnenko, these suggestions don't really have a precedent in the game when it comes to highlighting images aside from the army bar. I'm afraid it would immediately feel alien to long-time players. Also the game's GUI is generally very static: not lots of animation, glowing effects etc.

Perhaps a better solution would be to just use a grayscale filter on the icons of the resources that are lacking and color on the ones that are met?

As for buildings, I'm not sure. Gray vs white text wouldn't be very obvious. Red text is very unusual in the game. Generally there's only white and yellow, but these are too similar and would be too close to make sense imo.

**Comment by lisnenko** [comment]
@zenseii , I am a long-time player, since 1995. I feel that a red number of a particular resource would not look alien to me. It is impossible to make Heroes 2 better (quality of life improvements) without making some changes to the UI. I think people accept this fact.
Optionally we can make this number of a resource just bold without changing its color or accept your option with grayscale icons of resources.
Checkboxes, kindly mentioned by @DarthGandalf , are also an option, if not for resources, but for buildings.

**Comment by lisnenko** [comment]
@zenseii, another option would be to write the quantities of missing resources using a slash, e.g., "8/6", where 8 is the required amount of a resource and 6 is the amount currently available.

**Comment by lisnenko** [comment]
@zenseii "...I'm afraid it would immediately feel alien to long-time players..."
Did the introducing of the Turn Order dialog\rectangle during battles feel alien to long-time players?

**Comment by ihhub** [comment]
> [@zenseii](https://github.com/zenseii) "...I'm afraid it would immediately feel alien to long-time players..." Did the introducing of the Turn Order dialog\rectangle during battles feel alien to long-time players?

Sure. Please provide a solution to the problem including UX design and implementation.

**Comment by PusshPop** [comment]
I did some experimenting with text colors, and what I found is that with this font the color needs to be bright enough to contrast with the shadow. 
Therefor I dont think red is an option. The shadow of the letters makes a dark red text blurry (that or my eyes are deteriorating...) , and light red is not very appealing imo.
Grey could work, though it might not be ideal for people with color blindness and the evil interface. Also threw in black... a shade of grey 😄  

<img width="288" height="345" alt="Image" src="https://github.com/user-attachments/assets/464049e4-a5be-4763-975e-2bbeee035285" />
<img width="288" height="345" alt="Image" src="https://github.com/user-attachments/assets/f5b758c8-9cbb-47fc-9b4f-4ae82aafb541" />
<img width="288" height="345" alt="Image" src="https://github.com/user-attachments/assets/6a694ac8-968b-412b-adb9-6f8872218660" />
<img width="288" height="345" alt="Image" src="https://github.com/user-attachments/assets/9ab33362-01f6-47bd-bd23-56e084d8f0ae" />
<img width="288" height="345" alt="Image" src="https://github.com/user-attachments/assets/4ce4c9e0-6530-463e-b2ed-735c81fa43ce" />
<img width="288" height="345" alt="Image" src="https://github.com/user-attachments/assets/fb2de5eb-b2e3-4ee2-b413-907a662b0663" />
<img width="288" height="345" alt="Image" src="https://github.com/user-attachments/assets/8d68e66a-cd86-4b2c-a091-35a0ede14966" />
<img width="288" height="345" alt="Image" src="https://github.com/user-attachments/assets/b4952d95-95fc-4800-99c0-893e77c38eaf" />

**Comment by zenseii** [comment]
@PusshPop, another concern I have with the colors is that it's ambiguous what color/shade is unbuilt and what is built. How about adding a check mark ✔️ to the right of the building name? That is already what's done for buildings in the castle build screen, so that gives continuity. It's likely you can't just copy the checkmarks from the castle screen.

Another option that could be explored is having the small images of the buildings but I'm afriad the dialog would become too big.

**Comment by DarthGandalf** [comment]
Thanks for the experiment!

I can barely see that number 5 near crystal, in all these pictures. And I'm not colorblind

**Comment by lisnenko** [comment]
@PusshPop Please try dark red on both interfaces.
